addons/loyalty_delivery/tests/test_loyalty_delivery.py

```diff
@@ -71,17 +72,105 @@ def test_delivery_cost_gift_card(self):
+            })],
+        })
+        self.env['loyalty.generate.wizard'].with_context(active_id=program_ewallet.id).create({
+            'coupon_qty': 1,
+            'points_granted': 1000,
+        }).generate_coupons()
+        reward_ewallet = program_ewallet.reward_ids[0]
+        ewallet = program_ewallet.coupon_ids[0]
+
+        # Create an order and pay with the ewallet.
+        order = self.SaleOrder.create({
+            'partner_id': self.partner_1.id,
+            'pricelist_id': self.pricelist.id,
+            'order_line': [Command.create({
+                'product_id': self.product_4.id,
+                'price_unit': 200.00,
+            })]
+        })
+        order._apply_program_reward(reward_ewallet, ewallet)
 
         delivery_wizard = Form(self.env['choose.delivery.carrier'].with_context({
-            'default_order_id': sale_normal_delivery_charges.id,
-            'default_carrier_id': free_delivery.id,
+            'default_order_id': order.id,
+            'default_carrier_id': self.free_delivery.id,
         }))
         delivery_wizard.save().button_confirm()
 
-        self.assertEqual(len(sale_normal_delivery_charges.order_line), 3)
-        self.assertEqual(sale_normal_delivery_charges.amount_untaxed, 80.0, "Delivery cost is not Added")
+        self.assertEqual(order.order_line.filtered('is_delivery').price_total, 0)
+
+    def test_delivery_cost_discounts(self):
+        """
+            make sure discounts aren't taken into account for free delivery
+        """
+        discount90 = self.env['loyalty.program'].create({
+            'name': '90% Discount',
+            'program_type': 'coupons',
+            'applies_on': 'current',
+            'trigger': 'auto',
+            'rule_ids': [(0, 0, {})],
+            'reward_ids': [(0, 0, {
+                'reward_type': 'discount',
+                'discount': 90,
+                'discount_mode': 'percent',
+                'discount_applicability': 'order',
+            })]
+        })
+
+        # Create an order and apply discount.
+        order = self.SaleOrder.create({
+            'partner_id': self.partner_1.id,
+            'pricelist_id': self.pricelist.id,
+            'order_line': [Command.create({
+                'product_id': self.product_4.id,
+                'price_unit': 200.00,
+            })]
+        })
+        order._update_programs_and_rewards()
+        coupon = order.coupon_point_ids.coupon_id.filtered(lambda c: c.program_id == discount90)
+        order._apply_program_reward(discount90.reward_ids, coupon)
+        order.action_confirm()
+
+        delivery_wizard = Form(self.env['choose.delivery.carrier'].with_context({
+            'default_order_id': order.id,
+            'default_carrier_id': self.free_delivery.id,
+        }))
+        delivery_wizard.save().button_confirm()
+
+        self.assertEqual(order.order_line.filtered('is_delivery').price_unit, self.product_delivery_free.list_price)
```

Comment:
```suggestion
        self.assertEqual(
            order.order_line.filtered('is_delivery').price_unit, 
            self.product_delivery_free.list_price
        )
```

We try to stay under 100 chars =]